pkg/controller/jobframework/defaults.go

```diff
@@ -32,3 +34,17 @@ func ApplyDefaultForSuspend(job GenericJob, manageJobsWithoutQueueName bool) {
 		}
 	}
 }
+
+func ApplyDefaultLocalQueue(job GenericJob, defaultQueue string) {
+	if !features.Enabled(features.LocalQueueDefaulting) || defaultQueue == "" {
+		return
+	}
+	labels := job.Object().GetLabels()
+	if labels == nil {
+		labels = make(map[string]string, 1)
+	}
```

Comment:
I think we can avoid the allocation by moving this block below `if labels[constants.QueueLabel] == "" {`